### DISCUSSION TITLE
Can not get values from controlled Select component (getting null)

### DISCUSSION THREAD

**Discussion by dorin-js** [mantinedev/mantine#2103] [Q&A] [answered]
Can anybody help with onChange handler?
```jsx
function CategoryInput({
  collections, //list of categories from db
  category, //state from parent component
  setCategory, //setState
} {
  //options to be selected
  const values = collections.map((category) => ({
    value: category.catName,
    label: category.catName,
  }))
  return (
    <Select
      label="Category"
      placeholder="Pick one"
      searchable
      clearable
      required
      data={values}
      onChange={setCategory}
    />
  )
}
```

**Comment by rtivital** [comment, marked as answer]
See controlled Select example, you need to provide `value` and `onChange` props – https://mantine.dev/core/select/#controlled
Also, make sure that you are using latest version, there were some bug fixes for uncontrolled component in recent releases.

**Reply by dorin-js** [reply]
Oh..i see 😄 . i thought the problem is in onChange. I mean, it's a bit weird for me to not pass values to setState.
Thank you, sir. It worked!
